src/slopes.jl

```diff
@@ -0,0 +1,198 @@
+# Reference : Dietmar Ratz - An Optimized Interval Slope Arithmetic and its Application
+using IntervalArithmetic, ForwardDiff
+import Base: +, -, *, /, ^, sqrt, exp, log, sin, cos, tan, asin, acos, atan
+import IntervalArithmetic: mid, interval
+
+function slope(f::Function, x::Interval, c::Real)
+    try
+        f(slope_var(x, c)).s
+    catch y
+        if isa(y, MethodError)
+            ForwardDiff.derivative(f, x)
+        end
+    end
+end
+
+struct Slope{T}
+    x::Interval{T}
+    c::Interval{T}
+    s::Interval{T}
+    Slope{T}(a, b, c) where T = new(a, b, c)
+    Slope{T}(c) where T = Slope{T}(c, c, 0)
+end
+
+function slope_var(v::Real)
+    Slope{Float64}(v, v, 1)
+end
+
+function slope_var(v::Interval, c::Real)
+    Slope{Float64}(v, c, 1)
+end
+
+function interval(u::Slope)
+    u.x
+end
+
+function mid(u::Slope)
+    u.c
+end
+
+function slope(u::Slope)
+    u.s
+end
+
+function +(u::Slope, v::Slope)
+    Slope{Float64}(u.x + v.x, u.c + v.c, u.s + v.s)
+end
```

Comment:
Do not restrict the type to Float64 anywhere. Everything should be generic. You shouldn't need to specify that.

Reply:
In fact, rather you should add some tests using eg BigFloat.